recipe.py

```diff
@@ -32,21 +32,19 @@ def set_calories_per_portion(self, calories_per_portion:int):
     def get_ingredients(self):
         return self._ingredients
     
-    def set_ingredients(self,ingredients):
+    def set_ingredients(self,ingredients:list):
+        self._ingredients = ingredients
+
+    def add_ingredients(self,ingredients):
         self._ingredients.append(ingredients)
-        if len(set(self._ingredients)) != len(self._ingredients):
-            self.remove_ingredient(ingredients) #I have tested this and it only removes the recently added ingredient
-            return False
+        # if len(set(self._ingredients)) != len(self._ingredients):
```

Comment:
extend should be used if ingredients are presented as a list, it would best that the type be added in the call to make the point clear.